test/e2e/cli/pkg_unsource_test.go

```diff
@@ -0,0 +1,437 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package cli_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+
+	fcmd "kraftkit.sh/test/e2e/framework/cmd"
+	fcfg "kraftkit.sh/test/e2e/framework/config"
+)
+
+var _ = Describe("kraft pkg", func() {
+	var cmd *fcmd.Cmd
+
+	var stdout *fcmd.IOStream
+	var stderr *fcmd.IOStream
+
+	var cfg *fcfg.Config
+
+	BeforeEach(func() {
+		stdout = fcmd.NewIOStream()
+		stderr = fcmd.NewIOStream()
+
+		cfg = fcfg.NewTempConfig()
+
+		cmd = fcmd.NewKraft(stdout, stderr, cfg.Path())
+		cmd.Args = append(cmd.Args, "pkg")
+	})
+
+	_ = Describe("unsource", func() {
+		BeforeEach(func() {
+			cmd.Args = append(cmd.Args, "unsource", "--no-check-updates", "--log-level", "info", "--log-type", "json")
+		})
+
+		Context("unsourcing a link in the config file", func() {
+			BeforeEach(func() {
+				// Save the config file by moving it to a temporary location
+				err := os.Rename(cfg.Path(), cfg.Path()+".tmp")
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				// Restore the config file
+				err := os.Rename(cfg.Path()+".tmp", cfg.Path())
+				Expect(err).ToNot(HaveOccurred())
+			})
+			When("kraftkit was already present with a config file", func() {
+				BeforeEach(func() {
+					cmd.Args = append(cmd.Args, "https://example.com")
+					err := cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Replace the last argument
+					cmd.Args = cmd.Args[:len(cmd.Args)-1]
+
+					oldArgs := cmd.Args
```

Comment:
Changing the args here is a bit ambiguous because this `cmd` has run already. Maybe it would be more clear if we did a `copy(args, cmd.Args)` before altering the slice?